filecoin-proofs/tests/regression.rs

```diff
@@ -0,0 +1,265 @@
+use std::fs::File;
+use std::io::{self, BufRead};
+use std::path::Path;
+
+use anyhow::Result;
+use filecoin_proofs::{
+    verify_seal, SectorShape16KiB, SectorShape1GiB, SectorShape2KiB, SectorShape32GiB,
+    SectorShape32KiB, SectorShape4KiB, SectorShape512MiB, SectorShape64GiB, SectorShape8MiB,
+    SECTOR_SIZE_16_KIB, SECTOR_SIZE_1_GIB, SECTOR_SIZE_2_KIB, SECTOR_SIZE_32_GIB,
+    SECTOR_SIZE_32_KIB, SECTOR_SIZE_4_KIB, SECTOR_SIZE_512_MIB, SECTOR_SIZE_64_GIB,
+    SECTOR_SIZE_8_MIB,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+
+use filecoin_proofs::{Commitment, PoRepConfig, ProverId};
+#[cfg(feature = "persist-regression-proofs")]
+use filecoin_proofs::{
+    MerkleTreeTrait, SealCommitOutput, SealPreCommitOutput, PUBLISHED_SECTOR_SIZES,
+};
+use storage_proofs_core::sector::SectorId;
+
+const V16_SEAL_REGRESSION_RECORDS: &str = "seal_regression_records-v16.json";
+const V18_SEAL_REGRESSION_RECORDS: &str = "seal_regression_records-v18.json";
+
```

Comment:
Just an optional thought, but it may be a good idea to use a constant here
```rust
const SEAL_REGRESSION_RECORDS: [(&str, &str); 2] = [
    (V16_SEAL_REGRESSION_RECORDS, "v16"),
    (V18_SEAL_REGRESSION_RECORDS, "v18"),
];

fn run_seal_regression_tests() {
    ...
    for (path, version) in SEAL_REGRESSION_RECORDS {
        ...
    }
    ...
}
```
so that when new regression test files are added, only this files's `const`s need to be modified (rather than requiring both the `const`s and `fn run_seal_regression_tests` be modified).